server/bin/www

```diff
@@ -28,32 +29,73 @@ const server = http.createServer(app);
 /**
  * Listen on provided port, on all network interfaces, and sync database.
  */
-
 const io = require("socket.io")(server);
 
+
+//Authentication middleware for socket.io
+require('socketio-auth')(io, {
+  authenticate: validateSocket
+});
+
 io.on("connection", (socket) => {
+
+  //Create a private room for each user denoted by socketId. 
+  const {socketId, token} = socket.handshake.query;
+  socket.join(socketId);
```

Comment:
There are a couple concerns with this implementation:
1. Room name is passed-in from the client side so a malicious client can pass-in other user's room name to receive unauthorized socket events.
2. Room name is stored inside local storage and removed on logout. That's an additional sensitive information to manage on client side.

I suggest using user ID as a room name here.

Reply:
Those are really good points. Thank you!